src/scielo/bin/xml/prodtools/db/xc_models.py

```diff
@@ -51,6 +51,48 @@
 ])
 
 
+def get_journal_from_registered_title(registered_title):
+    """
+    Retorna os dados do periódico no objeto Journal a partir dos dados
+    obtidos da base de dados isis
+    """
+    journal = Journal()
+    journal.frequency = registered_title.frequency
+    journal.acron = registered_title.acron
+    journal.p_issn = registered_title.print_issn
+    journal.e_issn = registered_title.e_issn
+    journal.abbrev_title = registered_title.abbrev_title
+    journal.nlm_title = registered_title.journal_id_nlm_ta
+    journal.publisher_name = registered_title.publisher_name
+    journal.license = registered_title.license
+    journal.collection_acron = None
+    journal.journal_title = registered_title.journal_title
+    journal.issn_id = registered_title.issn_id
+    return journal
+
+
+def get_journal_data_from_registered_title(registered_title):
+    """
+    Retorna os dados do periódico no formato para fazer comparação com
+    os dados do documento XML
+    """
+    j_data = Journal()
+    j_data.acron = [registered_title.acron]
+    j_data.frequency = [registered_title.frequency]
+    j_data.p_issn = [registered_title.print_issn]
+    j_data.e_issn = [registered_title.e_issn]
+    j_data.abbrev_title = [registered_title.abbrev_title]
+    j_data.nlm_title = [registered_title.journal_id_nlm_ta]
+    j_data.publisher_name = [registered_title.publisher_name]
+    if isinstance(registered_title.publisher_name, list):
+        j_data.publisher_name = registered_title.publisher_name
+    j_data.license = [registered_title.license]
+    j_data.collection_acron = [None]
+    j_data.journal_title = [registered_title.journal_title]
+    j_data.issn_id = [registered_title.issn_id]
+    return j_data
+
+
```

Comment:
Dúvida @robertatakenaka, se `get_journal_data_from_registered_title` e `get_journal_from_registered_title` criam uma instância da mesma classe `Journal` porque os dados serão encapsulados dentro de uma lista?

Reply:
@joffilyfe o código precisaria ser refatorado um pouco mais, mas não agora. 
Em um dado ponto há uma [validação](https://github.com/scieloorg/PC-Programs/blob/6329dc10b62c74201600acbe3bbf793563aa8544/src/scielo/bin/xml/prodtools/validations/article_validations.py#L32) dos dados do periódico no XML vs dados registrados no SciELO que usa uma "lista de valores possíveis dos dados de periódico". O XC obtém os dados do periódico da base de dados para obter os dados, mas o XPM de um [arquivo CSV.](https://github.com/scieloorg/PC-Programs/blob/6329dc10b62c74201600acbe3bbf793563aa8544/src/scielo/bin/xml/prodtools/db/xc_models.py#L1511). O objeto Journal é instanciado com a lista de valores. Enfim, não há motivo especial nenhum, exceto de manter a compatibiliade com aquilo que já estava funcionando. Já que a correção era para tratar a exceção lançada por não encontrar os dados do fascículo.

Reply:
Obrigado pela explicação @robertatakenaka. Não vejo problema a duplicação do código, só queria entender mesmo. Recentemente vi um artigo que fala sobre isso, se te interessar: https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction

Reply:
@joffilyfe corrigido, por favor, verificar.